src/main/java/com/adyen/terminal/security/NexoCrypto.java

```diff
@@ -141,10 +141,9 @@ private void validateHmac(byte[] receivedHmac, byte[] decryptedMessage, NexoDeri
         if (receivedHmac.length != hmac.length) {
             valid = false;
         }
-        for (int i = 0; i < hmac.length && valid; i++) {
+        for (int i = 0; i < hmac.length; i++) {
             if (receivedHmac[i] != hmac[i]) {
```

Comment:
if `receivedHmac.length < hmac.length`
this will throw an ArrayIndexOutOfBoundsException

Reply:
maybe we can use the same method as we use for the cloud terminal api? 
https://github.com/Adyen/adyen-java-api-library/commit/3f903f432c6c11f7e7c3a09ea5ce4ace6b772183
`MessageDigest.isEqual(receivedHmac,hmac)`